### DISCUSSION TITLE
prisma and connection string from a promise

### DISCUSSION THREAD

**Discussion by Rahmo** [prisma/prisma#5932] [Q&A] [answered]
I have an edge case of setting prisma connection url from a promise. 
the reason why the connection comes from a promise is because all sensitive strings are stored in aws sim securely 
so I can call an endpoint that returns a promise then I should handle it here. 
then passing Prisma Client to a context in Apollo Server. 
the problem is the promise doesn't get resolved before its passed to Apollo 

for example I am doing this 
```ts
// example function 
function getURL(){
   return new Promise(resolve => setTimeout(() => resolve(process.env.DATABASE_URL), 3000));
}
```
```ts
// im setting a client like
const prisma = async ()=>{
  return new PrismaClient(
    {
      datasources:
        {
          db:
            {
              url: await getURL()
            }
        }
    }
  );
}
```

**Comment by ryands17** [comment, marked as answer]
@Rahmo 👋 
Could you share how you have passed this to Apollo and how are you using it i.e. Serverless or on a normal long-running server?

**Reply by Rahmo** [reply]
I got it resolved thanks

**Reply by Rahmo** [reply]
there was nothing wrong in the code above its just the promise before it gets there has errors

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
